### PR TITLE
Dual license the torch-mlir project.

### PR DESCRIPTION
**Pull request by silvasean**
**TL;DR: if you are ok with dual licensing your code contributions in torch-mlir, please respond "LGTM" in a comment below**

@ajarthurs @brycearden @byronyi @MATRIXKOO @marbre @mikeurbach @stephenneuendorffer @powderluv @dan-garvey @gpetters94 @harsh-nod @ramiro050 @raikonenfnu @saeta @phoenix-meadowlark @River707 @stellaraccident @cathyzhyi 

This commit (with approval from all contributors) dual licenses
the torch-mlir project under both the standard LLVM license and the
standard PyTorch license. This will facilitate moving code between
torch-mlir and the two upstream projects.

The standard file comment is now:

```
// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
// Also available under a BSD-style license. See LICENSE.
```

See `LICENSE` in the project root for the terms of both licenses.

### CODE REVIEW

**Comment by mikeurbach** [comment]
LGTM

**Comment by ramiro050** [comment]
LGTM

**Comment by powderluv** [comment]
LGTM

**Comment by phoenix-meadowlark** [comment]
LGTM

**Comment by cathyzhyi** [comment]
LGTM

**Comment by joker-eph** [comment]
LGTM

**Comment by saeta** [comment]
LGTM

**Comment by gpetters94** [comment]
LGTM

**Comment by River707** [comment]
LGTM

**Comment by byronyi** [comment]
LGTM

**Comment by dan-garvey** [comment]
LGTM

**Comment by stephenneuendorffer** [comment]
LGTM

**Comment by raikonenfnu** [comment]
LGTM

**Comment by harsh-nod** [comment]
LGTM

**Comment by stellaraccident** [comment]
LGTM

**Comment by MATRIXKOO** [comment]
LGTM

**Comment by brycearden** [comment]
LGTM

**Comment by ajarthurs** [comment]
LGTM

**Comment by silvasean** [comment]
We have LGTM from everyone except @marbre. I just checked the history and their contributions were to an earlier stage of the project -- none of that code ended up in the newly scoped torch-mlir project. So we are good to go!
